llm/memory.go

```diff
@@ -120,6 +120,14 @@ func EstimateGPULayers(gpus []discover.GpuInfo, f *ggml.GGML, projectors []strin
 	// add one layer worth of memory as a buffer
 	if blk0, ok := layers["blk.0"]; ok {
 		layerSize = blk0.Size()
+		//use layer with maximum size to avoid vram overflow
+		for i := 1; i < int(f.KV().BlockCount()); i++ {
+			if blkn, ok := layers[fmt.Sprintf("blk.%d", i)]; ok {
+				if blkn.Size() > layerSize {
+					layerSize = blkn.Size()
+				}
+			} 
+		}
 	} else {
 		slog.Warn("model missing blk.0 layer size")
 	}
```

Comment:
This can be simplified to always return the max layer size 

```suggestion
	// use layer with maximum size to avoid vram overflow
	layerSize = slices.MaxFunc(slices.Collect(maps.Values(layers)), func(a, b ggml.Layer) int {
		return cmp.Compare(a.Size(), b.Size())
	}).Size()
```

You'll also need to import `cmp`, `maps`, and `slices`